### ISSUE TITLE
Drawio on IE11

### ISSUE THREAD

**Issue by tdesloovere** [pawelrojek/nextcloud-drawio#22]
Hi,

we noticed that opening the drawio nextcloud app in IE11, the screen loads but the loading icon keeps rotating. Using Firefox in Windows, all seem to work ok. When directly browsing to www . draw .io in IE11, everything seems to be working as expected.

Is there a solution/workaround for this problem?

![screenshot ie11 drawio nextcloud](https://user-images.githubusercontent.com/18137793/32053288-2901bd0a-ba5b-11e7-9cd6-e521f5bedf9b.png)

greetings,
Tom

**Comment by giovannixyz** [comment]
We have the same problem.

**Comment by tavinus** [comment]
I can confirm this problem as well. 
I took a quick look at the code but could not pinpoint where this could be fixed or even debuged.
It works on Edge though.
PS: I don't use IE at all, since it sucks a lot.

**Comment by tdesloovere** [comment]
Indeed, IE11 sucks and is soon to be obsolete. Unfortunately, most of our users are still using IE11.

But good news, I think I found a solution/workaround for this issue. Apparently, IE11 does not support the Javascript includes() function described in /apps/drawio/js/editor.js.
I tried the solution of the user TecBrat (https://stackoverflow.com/questions/31119300/ie11-object-doesnt-support-property-or-method-includes-javascript-window) by adding:

 // IE does not support .includes() so I'm making my own:
  String.prototype.doesInclude=function(needle){
    return this.substring(needle) != -1;
  }
 
in the file /apps/drawio/js/editor.js just before the if statement "if (evt.data.length > 0 && origin.includes(evt.origin)) {"

Afterwards, I changed the origin.**includes**(evt.origin) to origin.**doesInclude**(evt.origin) and that seemed to do it.

I am by no means a Javascript guy, so I have no idea what I did exactly or what the impact may be. So, maybe someone with more knowledge than me should look at it, but as far as I can see, the drawio app kept on working in Firefox 57, Chromium 63.

greetings,

Tom

**Comment by tavinus** [comment]
Good job! :)

Is there any extra JS code already been loaded just for IE?
I mean, maybe one can just create the <string>.includes function if running on IE and then there is no need to even touch the other code.

If there is no dedicated IE being loaded, on can probably just add a conditional and create the include function inside the conditional.

I am not sure they care about IE though.
